### PR TITLE
Added an option to reorder pinned items in the sidebar

### LINKED ISSUE #2011: Option to sort pinned files (items)
**Is your feature request related to a problem? Please describe.**
I would like to be able to sort pinned items (on a left). Now I can pin them, but if I want to change the order of importance for me, I can't.

**Describe the solution you'd like**
be able to sort pinned files (items)

**Describe alternatives you've considered**
just that, option to sort pinned files. Windows file explorer can do, you can do also

**Additional context**

### PR DESCRIPTION
**Pull request by DanielSlincak**
Fixes #2011

### CODE REVIEW

**Comment by yaichenbaum** [comment]
@DanielSlincak Thank you for your pull request, here is some feedback after checking out the changes
- The pinned items order does not seem to be saved after reopening Files
- After reordering an item, the item stays highlighted as if I am still pressing down on it

File: Files/DataModels/SidebarPinnedModel.cs [hunk 1]
```diff
@@ -31,11 +67,18 @@ public void AddDefaultItems()
             Items.Add(AppSettings.VideosPath);
         }
 
+        /// <summary>
+        /// Gets the item sfrom the navigation page
```
Comment (Thomas1664) [inline Files/DataModels/SidebarPinnedModel.cs:71]: Can you fix the typo in this line?

Comment (DanielSlincak) [reply Files/DataModels/SidebarPinnedModel.cs:71]: Of course!

**Comment by DanielSlincak** [comment]
> @DanielSlincak Thank you for your pull request, here is some feedback after checking out the changes
> 
> * The pinned items order does not seem to be saved after reopening Files
> * After reordering an item, the item stays highlighted as if I am still pressing down on it

Thanks for the feedback, I'll look into it

**Comment by DanielSlincak** [comment]
@yair100 The problem with having the items stay highlighted is a bug within Microsoft.UI.Xaml, as can be seen in these issues: 

- [https://github.com/microsoft/microsoft-ui-xaml/issues/3131](https://github.com/microsoft/microsoft-ui-xaml/issues/3131)

- [https://github.com/microsoft/microsoft-ui-xaml/issues/3290](https://github.com/microsoft/microsoft-ui-xaml/issues/3290)

I haven't found a workaround that works yet, so I am not sure on how to continue with this.

Regarding the other notes, the new order is now being saved correctly and I removed the right-click options for sorting.

**Comment by yaichenbaum** [comment]
@DanielSlincak Thanks for the update, while I don't want to merge this in until the issue is fixed on the WinUI side, perhaps we can add this behind a setting that is off by default so that once it's fixed the code is already there and we can just enable it for everyone. @chingucoding Would you happen to know if this issue is being worked on from the WinUI side of things?

**Comment by chingucoding** [comment]
Given that no one is assigned nor a member of the community said it is being worked on, I would guess that that issue is not being worked on.

**Comment by yaichenbaum** [comment]
@DanielSlincak Can you solve the merge conflict?

**Comment by DanielSlincak** [comment]
@yair100 Yeah, I'll make some changes and then make another pull request with the changes. Hopefully by the end of the week, if that's okay.

**Comment by Thomas1664** [comment]
Maybe we should get rid of `LocationItem `and use `ListedItem` instead

**Comment by DanielSlincak** [comment]
@yair100 The sorting is now disabled by default and can be enabled by changing the setting in the experimental settings.

File: Files/Views/SettingsPages/Experimental.xaml [hunk 2]
```diff
@@ -38,6 +38,11 @@
                     Header="Show Owner in properties"
                     HeaderTemplate="{StaticResource CustomHeaderStyle}"
                     IsOn="{x:Bind AppSettings.ShowFileOwner, Mode=TwoWay}" />
+                <ToggleSwitch
+                    x:Uid="SettingsExperimentalSortPinnedFiItemsByDragging"
```
Comment (yaichenbaum) [inline Files/Views/SettingsPages/Experimental.xaml:42]: Can you add the strings for this in the resource files?

File: Files/Strings/en-US/Resources.resw [hunk 3]
```diff
@@ -1152,4 +1152,10 @@
   <data name="SettingsOpenItemsWithOneclick.Header" xml:space="preserve">
     <value>Open items with a single click</value>
   </data>
+   <data name="PinToSidebarByDraggingCaptionText" xml:space="preserve">
+    <value>Pin to Sidebar</value>
+  </data>
+  <data name="SettingsExperimentalSortPinnedFiItemsByDragging" xml:space="preserve">
```
Comment (yaichenbaum) [inline Files/Strings/en-US/Resources.resw:1158]:
```suggestion
  <data name="SettingsExperimentalSortPinnedFiItemsByDragging.Header" xml:space="preserve">
```

**Review by yair100 — APPROVED** [review]

**Comment by yaichenbaum** [comment]
@DanielSlincak can you take a look at #7625?

**Pull request merged**